eth/vm/base.py

```diff
@@ -374,15 +384,34 @@ class VM(BaseVM):
     """
 
     _state = None
+    _block = None
 
     def __init__(self, header: BlockHeader, chaindb: BaseChainDB) -> None:
         self.chaindb = chaindb
-        self.block = self.get_block_class().from_header(header=header, chaindb=self.chaindb)
+        self._initial_header = header
+
+    @property
+    def header(self) -> BlockHeader:
```

Comment:
What do you think about going ahead and converting these to methods.

```python
class VM:
    def get_header(self):
        ...
    def get_block(self):
        ...
```

Makes it a lot clearer that they have side effects.

Reply:
I am :+1: on it, but the API is used in a *lot* of places right now and I'd like to get this live for beam sync. So I'll add a good first issue.